### PR TITLE
Fix create account

### PR DESCRIPTION
**Pull request by kulmann**
This PR moves the service handler creation from "within the respective server" to the command which creates the two servers. This way we make sure to only have one index in memory. The accounts and groups handlers already have locking in place anyway, so code wise it already looks thread safe.

### CODE REVIEW

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/server.go  1
         

Complexity decreasing per file
==============================
+ pkg/server/http/server.go  -1
+ pkg/service/v0/service.go  -11
+ pkg/server/grpc/server.go  -1
         

Clones removed
==============
+ pkg/server/http/server.go  -1
+ pkg/server/grpc/server.go  -1
+ pkg/service/v0/groups.go  -2
+ pkg/service/v0/accounts.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6164139&bid=20132862)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/server.go  1
         

Complexity decreasing per file
==============================
+ pkg/service/v0/service.go  -11
+ pkg/server/http/server.go  -1
+ pkg/server/grpc/server.go  -1
         

Clones removed
==============
+ pkg/server/http/server.go  -1
+ pkg/service/v0/accounts.go  -2
+ pkg/service/v0/groups.go  -2
+ pkg/server/grpc/server.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6164139&bid=20132862)

File: pkg/service/v0/accounts.go [hunk 1]
```diff
@@ -4,6 +4,8 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	settings "github.com/owncloud/ocis-settings/pkg/proto/v0"
+	settings_svc "github.com/owncloud/ocis-settings/pkg/service/v0"
```
Comment (butonic) [inline pkg/service/v0/accounts.go:8]: your ide sorts this weirdly ... these are not part of the std lib ...

File: pkg/service/v0/service.go [hunk 2]
```diff
@@ -30,228 +32,58 @@ func New(opts ...Option) (s *Service, err error) {
...
-		} else if !fi.IsDir() {
-			return nil, fmt.Errorf("%s is not a directory", accountsDir)
-		}
+	// build an index
+	if s.index, err = s.buildIndex(); err != nil {
+		return nil, err
+	}
+	fmt.Print("\n\n\n")
+	fmt.Print(s.index)
+	fmt.Print("\n\n\n")
```
Comment (butonic) [inline pkg/service/v0/service.go:67]: nah, come on ...

**Review by butonic — CHANGES_REQUESTED** [review]

**Comment by butonic** [comment]
just nitpicking ... the actual change is sound!

File: pkg/service/v0/service.go (hunk 2 above)
Comment (kulmann) [reply pkg/service/v0/service.go]: That's super important code!

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/server.go  1
         

Complexity decreasing per file
==============================
+ pkg/server/http/server.go  -1
+ pkg/service/v0/service.go  -11
+ pkg/server/grpc/server.go  -1
         

Clones removed
==============
+ pkg/server/http/server.go  -1
+ pkg/service/v0/accounts.go  -2
+ pkg/service/v0/groups.go  -2
+ pkg/server/grpc/server.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6164139&bid=20132862)

File: pkg/service/v0/service.go (hunk 2 above)
Comment (kulmann) [reply pkg/service/v0/service.go]: removed it, thanks for spotting :-)

**Review by butonic — APPROVED** [review]
:rocket:

**Pull request merged**